Detectors/TRD/reconstruction/src/CruRawReader.cxx

```diff
@@ -484,6 +484,11 @@ int CruRawReader::processHalfCRU(int cruhbfstartoffset, int numberOfPreviousCRU)
 
   //get eventrecord for event we are looking at
   mIR.bc = mCurrentHalfCRUHeader.BunchCrossing; // correct mIR to have the physics trigger bunchcrossing *NOT* the heartbeat trigger bunch crossing.
+  //shift accordingly
+  mIR.bc = mCurrentHalfCRUHeader.BunchCrossing - o2::ctp::TriggerOffsetsParam::Instance().LM_L0;
+  if (mIR.bc < 0) {
+    mIR.bc = 0;
+  }
```

Comment:
Why setting BC to 0 ? The data corresponding to the BC which does not belong to TF should be simply discarded.

Reply:
good point, fixing.

Reply:
Also, the inverse (adding the bias) should be applied in the mc2raw.

Reply:
even better point ;-) I want to get the safety net fixed for the data decoder first, then will come back to this. I dont want to dump the entire dataframe rather only the halfcruheader in question, and with the codes current state its rather delicate.